### ISSUE TITLE
ZWESLSTC and ZWESISTC fail to start

### ISSUE THREAD

**Issue by hockeyrob** [zowe/community#1613]

**Comment by 1000TurquoisePogs** [comment]
I thought to place this issue into the ZSS category but it seems to be more general.
In the ZIS front, exit 8 is generic failure. I hope someone can shed more light on it but I'd like to know, is this your first time installing any zowe or is there another zowe already running on this system?

**Comment by hockeyrob** [comment]
This is the first time installing Zowe. I installed Node.js a few weeks ago, and then started the Zowe install, using the SMP/E-based install.

**Comment by hockeyrob** [comment]
I was able to try it with zwe internal start, and well down the output it said that NODE_HOME does not point to a valid install of Node. I ran node --version and got v16.13.0, so...is that too new? Would this be the problem with both ZWESLSTC and ZWESISTC?

**Comment by hockeyrob** [comment]
Upgraded to Zowe v2.2…zwe internal start allowed me to find that the Node.Js  path was incorrect…updating that allowed for it to progress a little…had to update the Zowe runtime directory then, and ZWESLSTC started grinding to life. Not fully alive yet…getting a lot of LE HEAP issues….and, with all these updates, ZWESISTC still won’t start. I’ll post the HEAP messages for the launcher.

Getting RACF messages about the ICSF routine calls, even though CSF specifies CHECKAUTH(NO). Does zowe do its own authorization checks before trying to do encryption? Or even CSFRNG calls?

I turned on the LE RPTSTG and RPTOPT options, to see why it was complaining about ALL31, HEAP and HEAP64…to no avail. Signal-to-noise ratio dropped dramatically.

And, ZWESISTC still fails, without any idea why.

**Comment by hockeyrob** [comment]
Okay, the last time I started ZWESLSTC I got 80 of these messages:

CEE3772I HEAP location set to BELOW forces HEAPPOOLS to be set OFF. 

Checking the RPTOPT output, I see:

Programmer default               HEAP(131072,524288,ANYWHERE,KEEP,16384,16384)

My specifications inside //CEEOPTS don't seem to affect this value, and regardless, you have ANY specified, so it shouldn't think HEAP is set for BELOW...unless the BELOWHEAP specification requires 24-bit HEAP allocation for the whole application.

Programmer default               BELOWHEAP(32768,16384,FREE)

Any suggestions on what I can try with ZWESISTC?

**Comment by hockeyrob** [comment]
Finally got something out of ZWESISTC:

ZWES0001I ZSS Cross-Memory Server starting, version is 2.2.0+20220713     
ZWES0002I Input parameters at 0x6FE6:                                     
ZWES0101I 00000000  0016D5C1 D4C57EE9 E6C5E2C9 E26DE2E3 |..NAME=ZWESIS_ST|
ZWES0101I 00000010  C46BD4C5 D47EF0F0                   |D,MEM=00|        
ZWES0004I Server name is 'ZWESIS_STD      '                               
ZWES0118E Core server started in wrong key 8                              
ZWES0011E ZSS Cross-Memory server not started, RC = 55                    
ZWES0013E ZSS Cross-Memory Server terminated due to an error, status = 8

No clue how to fix that one.

**Comment by hockeyrob** [comment]
ZWESISTC is working now, since I made 5ge SCHED00 update to make it nonswappable.
ZWESLSTC still not starting; will try your suggestion for PKCS12 certs.
R;

**Comment by 1000TurquoisePogs** [comment]
It looks like you found these parts of the doc or similar 
https://docs.zowe.org/stable/user-guide/configure-xmem-server#key-4-non-swappable 
https://docs.zowe.org/stable/user-guide/configure-uss#language-environment
https://docs.zowe.org/stable/user-guide/systemrequirements-zos#zwesvusr

But have you gotten new messages regarding PKCS12 that we can assist with?

**Comment by hockeyrob** [comment]
I’ve given up on it for now. I don’t know how to setup the certs for it, and can’t find enough info to educate myself. I want to set Zowe up with certs in OMVS first, and then e when it is working, I’ll get our security folks to create something in RACF like the OMVS setup so it works there too. Can’t do anything until I figure out how to create the cert setup in OMVS that Zowe needs.

R;

Rob Hamilton
Infrastructure Engineer
Chemical Abstracts Service

**Comment by hockeyrob** [comment]
No, I just don’t know how to set it up from scratch, what I need, where to put it, what has to happen to make it work, where to find the doc that explains that.

R;
